src/Runner.Worker/ActionDepthOutOfRange.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+
+namespace GitHub.Runner.Worker
+{
+    public class ActionDepthOutOfRange : Exception
+    {
+        public ActionDepthOutOfRange(int depth)
+            : base(FormatMessage(depth))
+        {
+        }
+
+        public ActionDepthOutOfRange(string message)
+            : base(message)
+        {
+        }
+
+        private static string FormatMessage(int depth)
+        {
+            return $"Composite action depth exceeded max depth {depth-1}, please simplify your actions.'";
```

Comment:
> `depth-1`

Should we just pass `maxDepth` instead to make this more intuitive and less fragile

Reply:
nit: consider dropping "please simplify your actions"

It may be no fault of the author. Might be the fault of a downstream action.